packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -399,22 +633,44 @@ class RenderParagraph extends RenderBox {
       if (_overflowShader != null) {
         // This layer limits what the shader below blends with to be just the text
         // (as opposed to the text and its background).
-        canvas.saveLayer(bounds, Paint());
+        context.canvas.saveLayer(bounds, Paint());
       } else {
-        canvas.save();
+        context.canvas.save();
       }
-      canvas.clipRect(bounds);
+      context.canvas.clipRect(bounds);
+    }
+    _textPainter.paint(context.canvas, offset);
+
+    RenderBox child = firstChild;
+    int childIndex = 0;
+    while (child != null) {
+      assert(childIndex < _textPainter.inlinePlaceholderBoxes.length);
+      final TextParentData textParentData = child.parentData;
```

Comment:
Why do we need to save this (and then retrieve it from) parentData? Isn't this just
```dart
Offset(
        _textPainter.inlinePlaceholderBoxes[childIndex].left,
        _textPainter.inlinePlaceholderBoxes[childIndex].top
      );
```

If we can get the scale (below) directly from the textPainter, why not this info as well?

Reply:
I guess, you still need the parentData stuff to be around for hitTesting. It is confusing here in paint that some of the information is retrieved directly from the text painter, and other is taken from the parent data. Unless we have a good reason, we should probably unify that here.